homeassistant/components/vallox/fan.py

```diff
@@ -202,3 +211,32 @@ async def async_turn_off(self, **kwargs: Any) -> None:
 
         # Same as for turn_on method.
         await self.coordinator.async_request_refresh()
+
+    async def async_set_percentage(self, percentage: int) -> None:
+        """Set the speed of the fan, as a percentage."""
+        if percentage == 0:
+            await self.async_turn_off()
+            return
+
+        update_needed = await self._async_set_percentage_internal(percentage)
+
+        if update_needed:
+            await self.coordinator.async_request_refresh()
+
+    async def _async_set_percentage_internal(self, percentage: int) -> bool:
+        """
+        Set fan speed percentage for current profile.
+
+        Returns true if speed has been changed, false otherwise.
+        """
+        if percentage == self.percentage:
```

Comment:
Does it hurt the device to set the same speed again? We normally don't guard against that case in polling integrations as a guard makes it impossible to set a different state if we have the wrong local state.

Reply:
I removed this check.